compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java

```diff
@@ -106,17 +110,92 @@ private static ContractABI buildABI(NeoModule neoModule) {
     }
 
     private static List<String> buildSupportedStandards(ClassNode asmClass) {
-        Optional<AnnotationNode> annotationNode = getAnnotationNode(asmClass,
-                SupportedStandards.class);
-        if (!annotationNode.isPresent()) {
-            return new ArrayList<>();
+        return getAnnotationNode(asmClass, SupportedStandards.class)
+                .flatMap(ManifestBuilder::transformAnnotationNodeStringValue)
+                .orElse(new ArrayList<>());
+    }
+
+    private static List<ContractGroup> buildGroups(ClassNode asmClass) {
+        return getAnnotationNode(asmClass, Groups.class)
+                .map(ManifestBuilder::getContractGroups)
+                .orElse(new ArrayList<>());
+    }
+
+    public static List<ContractPermission> buildPermissions(ClassNode asmClass) {
+        return getAnnotationNode(asmClass, Permissions.class)
+                .map(ManifestBuilder::getContractPermissions)
+                .orElseGet(() -> {
+                    ContractPermission contractPermission = new ContractPermission("*",
+                            Collections.singletonList("*"));
+                    return Collections.singletonList(contractPermission);
+                });
+    }
+
+    private static List<String> buildTrusts(ClassNode asmClass) {
+        return getAnnotationNode(asmClass, Trusts.class)
+                .map(ManifestBuilder::getContractTrusts)
+                .orElse(new ArrayList<>());
+    }
+
+    private static Optional<List<String>> transformAnnotationNodeStringValue(
+            AnnotationNode annotationNode) {
+
+        return ofNullable(annotationNode)
+                .map(ann -> {
+                    List<String> values = new ArrayList<>();
+                    for (Object value : (List<?>) ann.values.get(1)) {
+                        values.add((String) value);
+                    }
+                    return values;
+                });
+    }
+
+    private static List<ContractPermission> getContractPermissions(AnnotationNode ann) {
```

Comment:
We could enforce rules on the annotation values here.
The contract value has to be either a 33 (public key) or 20 bytes long (script hash) hex string.

Reply:
Yet another one. 😄 
I will implement it.

Reply:
implemented.